cpp/src/jni/CMakeLists.txt

```diff
@@ -18,10 +18,6 @@
 #
 # arrow_jni
 #
-if(ARROW_ORC)
-  add_subdirectory(orc)
```

Comment:
I think we'll want to keep the `if(ARROW_ORC)` (and move it to the other file)

Reply:
At the end all cpp/src/jni directory will be deleted

Reply:
That's not relevant, right? We only want to build ORC support if ORC is actually enabled.

Though, what is the intended new structure? Larry's PR in #13481 shows that we need to update CI/docs as well?

Reply:
What I mean is: how are we ensuring we only build ORC when ORC is actually enabled in the new structure?

Reply:
Oh ok clear, let me add that logic on `java/adapter/orc/src/main/cpp` not in `cpp/src/jni/CMakeLists.txt`

Related to the intended: For the ticket related to “Consolidate JNI compilation ORC ARROW-16930 / Dataset ARROW-16941”. Currently: ORC JNI / Dataset JNI are inside of ./cpp directory

Changes to Consolidate JNI compilation ORC / Dataset:

1st Phase:
1. Move ORC JNI / Dataset JNI from ./cpp to ./java directory
2. Delete ORC JNI / Dataset JNI from ./cpp
2. Continue building ORC / Dataset from ./cpp as you see at https://github.com/apache/arrow/blob/6bd03d409a687011b8d4b66e491015128f1ae27e/ci/scripts/java_jni_macos_build.sh#L52:L87 with the aggregation that this is using now ./java directory instead of cpp/src/jni directory

2nd Phase:
1. Building ORC / Dataset directly from ./java directory as a maven project “mvn clean install ...”
This phase involve lots of changes Java code + CMakeList + CI CD changes

Phase 1 will be implemented by this PR: Dataset (https://github.com/apache/arrow/pull/13481) / ORC (https://github.com/apache/arrow/pull/13458)

Phase 2: We are going to create ticket and maintain that on the backlog

Reply:
Added ARROW_ORC parameter